servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java

```diff
@@ -3779,6 +4182,38 @@ boolean auditLogRoleMember(StringBuilder auditDetails, RoleMember roleMember, bo
         return firstEntry;
     }
 
+    void auditLogGroupMembers(StringBuilder auditDetails, String label,
+                             Collection<GroupMember> values) {
+        auditDetails.append(", \"").append(label).append("\": [");
+        boolean firstEntry = true;
+        for (GroupMember value : values) {
+            firstEntry = auditLogGroupMember(auditDetails, value, firstEntry);
+        }
+        auditDetails.append(']');
+    }
+
+    boolean auditLogGroupMember(StringBuilder auditDetails, GroupMember groupMember, boolean firstEntry) {
+        firstEntry = auditLogSeparator(auditDetails, firstEntry);
+        auditDetails.append("{\"member\": \"").append(groupMember.getMemberName()).append('"');
+        if (groupMember.getExpiration() != null) {
+            auditDetails.append(", \"expiration\": \"").append(groupMember.getExpiration().toString()).append('"');
+        }
+        auditDetails.append(", \"approved\": ");
+        auditDetails.append(auditLogBooleanDefault(groupMember.getApproved(), Boolean.FALSE));
+        auditDetails.append(", \"system-disabled\": ");
+        auditDetails.append(groupMember.getSystemDisabled() == null ? 0 : groupMember.getSystemDisabled());
+        auditDetails.append("}");
+        return firstEntry;
+    }
+
+    String auditLogBooleanDefault(Boolean value, Boolean defaultValue) {
```

Comment:
I might be missing something but shouldn't this be reversed?
     if (defaultValue == Boolean.TRUE) {
            return value == Boolean.FALSE ? "false" : "true"; // in any other case that it isn't false - set to true
        } else {
            return value == Boolean.TRUE ? "true" : "false"; // in any other case that it isn't true - set to false
        }

Reply:
the logic is correct but I renamed the defaultValue to checkValue to be more clear the usage. the unit test has more explanation